src/debug-integration.adoc

```diff
@@ -5,6 +5,69 @@ This section describes changes to integrate the Sdext ISA and
 {cheri_base_ext_name}. It must be implemented to make external debug compatible
 with {cheri_base_ext_name}. Modifications to Sdext are kept to a minimum.
 
+The following features, which are optional in Sdext, must be implemented for
+use with {cheri_base_ext_name}:
+
+* The `hartinfo` register must be implemented.
+* All harts which support {cheri_base_ext_name} must provide
+  `hartinfo.nscratch` of at least 1 and implement the <<dscratch0c>> register.
+* All harts which support {cheri_base_ext_name} must provide
+  `hartinfo.datasize` of at least 1 if `hartinfo.dataaccess` is 0,
+  `hartinfo.datasize` at least `XLENMAX/32` otherwise.
+* The program buffer must be implemented, with `abstractcs.progbufsize` of at
+  least 4 if `dmstatus.impebreak` is 1, or at least 5 if `dmstatus.impebreak`
+  is 0.
+* The <<pcc>> must be readable using an <<AUIPCC>> instruction while executing
+  from the program buffer.
```

Comment:
Why is this necessary?

Reply:
Every argument I can think of for forbidding reads of pcc via AUIPCC applies just as strongly to csrr of pcc, which I why I tried to move away from using pcc at all as the sole source of Infinity in #44.

Reply:
Split that out into #90 to make this a smaller change.

Reply:
Thanks for splitting that out. Can we remove this bullet-point altogether? Reading the debug spec, support for CSR instructions is not optional when executing code from the program buffer.

Reply:
§4.1

> 12. Instructions that depend on the value of the PC (e.g. auipc) may act as illegal instructions.

I think that "csrr rd, pcc" is an "instruction that depends on the value of the PC".

Reply:
This PR was merged, so there is no longer a PCC CSR: https://github.com/riscv/riscv-cheri/pull/113.